File: habitual/backend/habitual/schemas/entries.py
from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel


class EntryCreate(BaseModel):
    content: str


class EntryRead(BaseModel):
    id: UUID
    timestamp: datetime
    content: str


class EntryUpdate(BaseModel):
    content: Optional[str] = None


class EntryListResponse(BaseModel):
    items: List[EntryRead]
    total: int
    limit: int
    offset: int

File: habitual/frontend/app/types/entries.ts
export type EntryRead = {
  id: string;
  timestamp: string; // ISO datetime string
  content: string;
};

export type EntryListResponse = {
  items: EntryRead[];
  total: number;
  limit: number;
  offset: number;
};

export type EntryCreate = {
  content: string;
};

export type EntryUpdate = {
  content?: string;
};

